snuba/subscriptions/scheduler_processing_strategy.py

```diff
@@ -437,7 +444,14 @@ def submit(self, message: Message[CommittableTick]) -> None:
         tasks = self.__schedulers[tick.partition].find(tick)
 
         start_encoding = time.time()
-        encoded_tasks = [self.__encoder.encode(task) for task in tasks]
+
+        if self.__processes > 1:
+            encoded_tasks = self.__multiprocessing_pool.map(
+                self.__encoder.encode, tasks
```

Comment:
This is very unlikely to give you any improvements framed this way:
Every subscription will be pickled, then send on a socket to the process, then unpickled and same for the response. 
The pickling unpickling and lock contention will likely make things worse. 

A way you can try is to batch them into relatively small groups of messages and send batches to each process. 
More effective would be the idea of using the parallel transform step we have in arroyo that avoid the communication via socket. 

Or alternative. Send a tick to a process. The process loads the the tasks and generates all the subscriptions for that tick.

Fundamentally the idea is to reduce the inter process communication and increase the size of the task each process has to do every time. With threads things would be different (to an extent) because they all share the same memory so they can access the same data structures, but processes cannot.